pkg/knowledge_base/edge_kb.go

```diff
@@ -136,7 +140,8 @@ func (kb EdgeKB) findPaths(source reflect.Type, dest reflect.Type, stack []Edge,
 			}
 		}
 		for _, e := range kb.GetEdgesWithTarget(source) {
-			if e.Destination == source && !visited[e.Source] && kb.isValidForPath(e, dest) {
+			det, _ := kb.GetEdgeDetails(e.Source, e.Destination)
+			if det.ReverseDirection && e.Destination == source && !visited[e.Source] && kb.isValidForPath(e, dest) {
```

Comment:
nit: This and the previous could use a little summary comment explaining the directionality and why (use cases) it can be reversed.